src/tests/smart_contract_test.ts

```diff
@@ -55,6 +58,118 @@ describe("SmartContract", () => {
     });
   });
 
+  describe(".register", () => {
+    const networkId = erc20ExternalModel.network_id;
+    const contractName = erc20ExternalModel.contract_name;
+    const contractAddress = erc20ExternalModel.contract_address;
+
+    it("should register a new smart contract", async () => {
+      Coinbase.apiClients.smartContract = smartContractApiMock;
+      Coinbase.apiClients.smartContract.registerSmartContract = jest
+        .fn()
+        .mockResolvedValue({ data: erc20ExternalModel });
+
+      const smartContract = await SmartContract.register(
+        networkId,
+        contractAddress,
+        testAllReadTypesABI,
+        contractName,
+      );
+
+      expect(Coinbase.apiClients.smartContract!.registerSmartContract).toHaveBeenCalledWith(
+        networkId,
+        contractAddress,
+        {
+          abi: JSON.stringify(testAllReadTypesABI),
+          contract_name: contractName,
+        },
+      );
+      expect(smartContract).toBeInstanceOf(SmartContract);
+      expect(smartContract.getContractAddress()).toBe(contractAddress);
+    });
+
+    it("should throw an error if creation fails", async () => {
+      Coinbase.apiClients.smartContract!.registerSmartContract = jest
+        .fn()
+        .mockRejectedValue(new Error("Failed to register the smart contract"));
+      await expect(
+        SmartContract.register(networkId, contractAddress, testAllReadTypesABI, contractName),
+      ).rejects.toThrow("Failed to register the smart contract");
+    });
+  });
+
+  describe(".update", () => {
```

Comment:
Can we fast follow with tests for when name or ABI are omitted, since they are optional?

Reply:
Yep, I'll have a follow up diff to address it!